File: BlogLite-Server/src/main/java/com/wsmrxd/bloglite/Utils/HttpUtil.java
package com.wsmrxd.bloglite.Utils;

import javax.servlet.http.HttpServletRequest;

import java.util.Objects;

public class HttpUtil {

    public final static String PROXY_PASSED_IP_HEADER = "X-Real-IP";

    public static String getIP(HttpServletRequest request) {
        return Objects.requireNonNullElse(
                request.getHeader(PROXY_PASSED_IP_HEADER),
                request.getRemoteAddr());
    }
}

File: BlogLite-Server/src/main/java/com/wsmrxd/bloglite/utils/HttpUtil.kt
package com.wsmrxd.bloglite.utils;

import javax.servlet.http.HttpServletRequest

const val PROXY_PASSED_IP_HEADER = "X-Real-IP"

fun getIPFromRequest(request: HttpServletRequest): String {
    return request.getHeader(PROXY_PASSED_IP_HEADER) ?: request.remoteAddr
}